lib/clientstate.js

```diff
@@ -37,14 +37,14 @@ export class ClientStateCompilerOptions {
     }
 
     fromJsonData(jsondata) {
-        if (typeof jsondata.binary !== 'undefined') this.binary = jsondata.binary;
```

Comment:
Not entirely sure, but I think I did this for a reason. But it might be that this alternative version will work..

Reply:
I _think_ oldschool js made you do it this way. Lemme google a bit.

Reply:
TLDR "typeof" works with undeclared variables (which isn't this case here, we have a declaration): https://thisthat.dev/variable-undefined-vs-typeof-variable-undefined/
So I think we're ok here?

Reply:
I guess I can check out the eslint thing too

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1492 seems to be the thing?

Reply:
AH, no they can be undefined because jsondata will be uncontrolled user json

Reply:
As in, this should not throw but be flexible and everything optional. Unless there's a different way of expressing that, I think that lint option should be disabled?

Reply:
> AH, no they can be just "not there" because jsondata will be uncontrolled user json
Not there in a named variable is different from "a variable that hasn't been defined"

Reply:
as discussed in discord, this is totally fine